crates/bevy_ecs/src/world/mod.rs

```diff
@@ -175,6 +175,7 @@ impl World {
         assert_eq!(ON_INSERT, self.init_component::<OnInsert>());
         assert_eq!(ON_REPLACE, self.init_component::<OnReplace>());
         assert_eq!(ON_REMOVE, self.init_component::<OnRemove>());
+        assert_eq!(UNIT, self.init_resource::<()>());
```

Comment:
I was following the convention here of asserting the component id is what we expect.